hail/python/hail/table.py

```diff
@@ -304,22 +304,30 @@ class Table(ExprContainer):
     """
 
     @staticmethod
-    def _from_ir(table_ir):
-        jir = Env.hail().expr.Parser.parse_table_ir(str(table_ir))
-        return Table(Env.hail().table.Table(Env.hc()._jhc, jir))
+    def _from_java(jt):
+        return Table(JavaTable(jt.tir()))
 
-    def __init__(self, jt):
+    def __init__(self, tir):
         super(Table, self).__init__()
 
-        self._jt = jt
+        self._tir = tir
+
+        r = Renderer(stop_at_jir=False)
+        code = r(tir)
+        env = {name: ir._jir for name, ir in r.jirs.items()}
+
+        self._jtir = Env.hail().expr.Parser.parse_table_ir(code, env)
+        self._jt = Env.hail().table.Table(Env.hc()._jhc, self._jtir)
```

Comment:
Ah, OK, just one change here. We talked about the reparse problem yesterday?  I had added stop_at_jir to fix it.  Namely, IRs can have an `_jir` attribute.  If they do, and stop_at_jir=True, then we don't render the IR, we just render a JavaMatrix/Table referring to that _jir and stop.

So I think you should use stop_at_jir=True, and set `tir._jir = _jtir`.

Reply:
Doesn't this do quadratic parsing? Can we fix that by setting the `_jir` on `tir` to `self._jtir`?

Reply:
Yes, we're in agreement.

Reply:
I didn't read the last line of your comment, sorry! :( today is a bit frantic

Reply:
once we build up IR types in Python we can totally get rid of the incremental parsing too

Reply:
Yes.

We also need to make the core Python code JVM-free.  Only the SparkBackend should call into the JVM.  Apart from this, we need to get the jtype out of the HailType (ptypes are needed for that) and then all the utilities we call into the JVM for (string escaping, etc.)

Reply:
I didn't 100% follow what to change. Can you double check I got what you described correctly?